### PR TITLE
Objective order fix

### PR DESCRIPTION
**Pull request by itamargreen**
There is a catch:

for this to work, the `id` of mission objectives in future challenges' `xml` files has to follow this format:

    <any-text>-<objective index>

the `-` hyphen is important, and the index must match by the order of objectives in each mission. 

The changes in these commits are backwards compatible, which means that older scoresheets still work.

### CODE REVIEW

File: tools/buildchallenge.js [hunk 1]
```diff
@@ -149,3 +150,41 @@ function processFile(fn) {
         // console.log(json);
     });
 }
+
+function checkOrder(missions) {
+    var needsChange = false;
```
Comment (idanstark42) [inline tools/buildchallenge.js]: Is this veriable for all the missinos together, or for each mission seperataly?

File: tools/buildchallenge.js [hunk 2]
```diff
@@ -132,6 +132,7 @@ function processFile(fn) {
             missions: missions.map(parseMission),
             strings: strings
         };
+        checkOrder(jsm.missions);
```
Comment (idanstark42) [inline tools/buildchallenge.js]: If you check the validation of each mission, shouldn't this be a part of the parseMission function, rather then it's own function?

Comment (itamargreen) [reply tools/buildchallenge.js]: probably. I worked under the idea of minimal change to existing code. That way it's easier to know what was already there. I agree that it should be moved. Will do now.

Comment (itamargreen) [reply tools/buildchallenge.js]: @idanstark42  now that I think about it, the idea is to verify that the objectives in each mission are ordered correctly **after** parsing every mission, but before creating the html files and js files and such.

**Comment by itamargreen** [comment]
by the way, the "changeNeeded" variable and the entire check is not so important. in missions that are already ordered correctly, the "reordering" won't actually change anything. I checked the impact on run time (difference in execution time) with and without the order check and it is about 1 millisecond difference.

**Comment by idanstark42** [comment]
I don't think I still understand the goal of this PR. Can you write in a sentence or two what it is?

**Pull request closed without merging**